sycl/doc/MultiTileCardWithLevelZero.md

```diff
@@ -0,0 +1,125 @@
+# Considerations for programming to multi-tile and multi-card under Level-Zero backend
+
+## 1 Devices discovery
+### 1.1 Root-devices
+
+Intel GPUs are represented as SYCL GPU devices, root-devices.
+The discovery of root-devices is best with "sycl-ls" tool, for example:
+	
+```
+$ sycl-ls
+[opencl:0] GPU : Intel(R) OpenCL HD Graphics 3.0 [21.19.19792]
+[opencl:0] CPU : Intel(R) OpenCL 2.1 [2020.11.11.0.03_160000]
+[level_zero:0] GPU : Intel(R) Level-Zero 1.1 [1.1.19792]
+[host:0] HOST: SYCL host platform 1.2 [1.2]
+```
+
+Note that "sycl-ls" shows all devices from all platforms of all SYCL backends that are seen
+by SYCL runtime. Thus in the example above there is CPU (managed by OpenCL backend) and 2!
+GPUs corresponding to the single physical GPU (managed by either OpenCL or Level-Zero backend).
+There are few ways to filter observable root-devices.
+	
+One is using environment variable SYCL_DEVICE_FILTER described in [EnvironmentVariables.md](https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md)
+```
+$ SYCL_DEVICE_FILTER=level_zero sycl-ls
+[level_zero:0] GPU : Intel(R) Level-Zero 1.1 [1.1.19792]
+```
+Another way is to use similar SYCL API described in [FilterSelector.md](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/FilterSelector/FilterSelector.adoc)
+E.g. `filter_selector("level_zero")` will only see Level-Zero operated devices.
+
+If there are multiple GPUs in a system then they will be seen as multiple different root-devices.
+On Linux these would be multiple SYCL root-devices of the same SYCL platform (representing Level-Zero driver).
+On Windows these would appear as root-devices of multiple different SYCL platforms (Level-Zero drivers).
+
+`CreateMultipleRootDevices=N NEOReadDebugKeys=1` evironment variables can be used to emulate multiple GPU cards.
+	
+### 1.2 Sub-devices
+	
+Some Intel GPU HW is composed of multiple tiles, e.g. 4 tile ATS.
+The root-device in such cases can be partitioned to sub-devices, each corresponding to the physical tiles.
+
+``` C++	
+try {
+  vector<device> SubDevices = RootDevice.create_sub_devices<
+  cl::sycl::info::partition_property::partition_by_affinity_domain>(
+  cl::sycl::info::partition_affinity_domain::next_partitionable);
+}
+```
+
+Each call to `create_sub_devices` will return exactly the same sub-devices and in the persistent order.
+To control what sub-devices are exposed by Level-Zero UMD one can use ZE_AFFINITY_MASK environment variable.
+
+NOTE: The `partition_by_affinity_domain` is the only partitioning supported for Intel GPU.
+Similar `next_partitionable` and `numa` are the only partitioning properties supported (both doing the same thing).
+
+`CreateMultipleSubDevices=N NEOReadDebugKeys=1` environment variables can be used to emulate multiple tiles of a GPU.

```

Comment:
same here, some example might be helpful.

Reply:
Today "sycl-ls" doesn't print info about sub-devices, so I can't give short example